implementations/rust/ockam/ockam_entity/src/imp.rs

```diff
@@ -207,6 +207,10 @@ impl<V: ProfileVault> ProfileContacts for ProfileImpl<V> {
     }
 
     fn verify_and_add_contact(&mut self, contact: Contact) -> ockam_core::Result<()> {
+        // Don't allow a profile to be a contact of itself.
+        if &self.identifier == contact.identifier() {
```

Comment:
@SanjoDeundiak what do you think of this check? I thought of the case when writing tests and saw I was able to add a profile as a contact of itself. Seems like it could cause some problems in the future.

Reply:
@jared-s I would say self contact should in contacts list by default. Therefore, adding it will cause duplicate error